core/src/cli_commands/bootstrap.rs

```diff
@@ -42,15 +47,14 @@ use crate::{
 /// minted tokens.
 pub fn bootstrap(yaml_config_file_name: String) -> Result<(), BootstrapError> {
```

Comment:
What's the plan for providing the bootstrap config template? With added contract parameters, it's very hard to manually craft. What if we add `--dump-bootstrap-config-template` option and generate it from some default bootstrap config values with placeholders for token ids, etc.?  
Although it's largely out of the scope of this PR, our bootstrap command will be effectively broken if we merge it as is.

Reply:
If we add the `--dump-bootstrap-config-template` option, we'll need to hard code the values in, so on any change we'll need to push out a new version of the app.

How about using the existing `bootstrap_config_example.yaml` file in the project root, where we put in all the default values. We give an instruction in the README to use it as a base for bootstrap. Perhaps even apply a SHA256 hash on the file and prominently show it on the README file. What do you reckon?

Reply:
> If we add the `--dump-bootstrap-config-template` option, we'll need to hard code the values in, so on any change we'll need to push out a new version of the app.

Not necessarily. One can craft any bootstrap config with any (reasonable) contracts. It's just a suggestion. Using this option is optional. :)

> 
> How about using the existing `bootstrap_config_example.yaml` file in the project root, where we put in all the default values. We give an instruction in the README to use it as a base for bootstrap. Perhaps even apply a SHA256 hash on the file and prominently show it on the README file. What do you reckon?

In this case, we will have contracts encoded twice (tests + config example). Plus, in the config example, there will be a chance of someone accidentally misses/deletes one char leading to a crash in runtime not covered by our tests. I'd rather have one source of truth for contracts in the source code.

Reply:
I've implemented the following solution: provide 2 types of bootstrap commands, regular and advanced. Most operators will use the regular one, which uses the default contract parameters. They just need to explicitly set the values described in the [EIP](https://github.com/ergoplatform/eips/blob/196e89a8f98bc1611473f059a9e58b81ca7d18d2/eip-0023.md#prerequisites). I've updated `/bootstrap_config_example.yaml`.

The advanced version allows the operator to configure everything. Also, everything is encoded once in `default_parameters.rs`.

Reply:
I'm upping the bounty since this rabbit hole is getting deeper. :)

What benefits do you see with two types of bootstrap commands? I thought of the following workflow when I proposed the `dump-bootstrap-config-template` option:
1. Don't include any `bootstrap_config_example.yml` in the binary distribution.
2. In the description(help) for `Bootstrap` command add that the config file can be generated with `--dump-bootstrap-config-template` option.
3. The dumped config file is a serialized `BootstrapConfig` struct filled with default values (contracts, token params, etc.).
4. The operator modifies the config file to their liking (including contracts).
5. Run the bootstrap command with the modified config file.

I see the following benefits vs. two bootstrap commands:
- No additional overhead for an operator to know two bootstrap commands.
- No need to maintain `bootstrap_config_example.yml` (i.e., update when the code is changed).
- Simplified code with only one bootstrap command.

BTW, the existing contract's P2S addresses in the code have mainnet network prefix, and it will not change in the future, so there is no need to pass the network prefix parameter.

Reply:
Thank you for your detailed reply! I totally agree with your reasoning. I missed the part in your explanation to generate the bootstrap yaml; I read '...default bootstrap config values with...' and automatically assumed the existence of a template file.

I included 2 separate commands because I'm assuming (perhaps wrongly) that most operators will not touch the contract values at all, and adding it to the bootstrap yaml will add unnecessary noise. It's not enough of a reason to go with this approach though. I'll implement your proposal and thank you again for the bounty increase!

Reply:
> 
> I included 2 separate commands because I'm assuming (perhaps wrongly) that most operators will not touch the contract values at all, and adding it to the bootstrap yaml will add unnecessary noise. It's not enough of a reason to go with this approach though. I'll implement your proposal and thank you again for the bounty increase!

You are right about the noise. Most operators will not touch the contract values. On the bright side - the contracts will be explicitly defined in the config files from the start with everyone be able to check them.

Reply:
I've made the necessary changes and I've also removed the network prefix from the contract P2S addresses. Since this all came about from my misunderstanding of your proposal, I think the last bounty increase isn't warranted. I'm good if it is reverted to the prior amount.

Reply:
No worries, the bootstrap config was out of the scope of the original issue, so a bounty increase was totally warranted. I appreciate your concerns.